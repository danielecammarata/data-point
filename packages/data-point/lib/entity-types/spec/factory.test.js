/* eslint-env jest */

const factory = require("./factory");

const EntityEntry = require("../entity-entry");

describe("reducer/reducer-entity-id#create", () => {
  const stub = () => {};
  test("throw if factory is not a function", () => {
    expect(() => {
      factory.create({}, "entry:abc");
    }).toThrowErrorMatchingSnapshot();
  });
  test("throw if resolve has wrong arity", () => {
    expect(() => {
      factory.create(stub, "entry:abc");
    }).toThrowErrorMatchingSnapshot();
  });
  test("entry", () => {
    const reducer = factory.create(EntityEntry, "entry:abc");
    expect(EntityEntry).toBe(reducer);
  });
});
