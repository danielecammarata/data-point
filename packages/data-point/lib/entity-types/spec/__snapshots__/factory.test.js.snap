// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducer/reducer-entity-id#create throw if factory is not a function 1`] = `"Entity Factory 'entry:abc' should be a function: {}"`;

exports[`reducer/reducer-entity-id#create throw if resolve has wrong arity 1`] = `"Entity Factory 'entry:abc' method should have an arity of 2, instead got: 0"`;
